### PR TITLE
[Fix][Transaction] Fix transaction pendingAckStore asyncMarkDelete

### PR DESCRIPTION
**Pull request by liangyepianzhou**
### Motivation
When we only use transaction to ack message, a message per transaction,
`cursor.asyncMarkDelete` will never be executed, because firstPosition and
 deletePosition both are init as metadataPositions.firstEntry().getKey();
 And if only the firstkey need to delete,  deletePosition also get metadataPositions.firstKey(); irstPosition == deletePosition, never call `cursor.asyncMarkDelete`
### Modification
init firstPosition and deletePosition as PositionImpl.EARLIEST

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by congbobo184 — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/MLPendingAckStore.java [hunk 1]
```diff
@@ -241,8 +241,8 @@ public void addComplete(Position position, ByteBuf entryData, Object ctx) {
                 completableFuture.complete(null);
 
                 if (!metadataPositions.isEmpty()) {
-                    PositionImpl firstPosition = metadataPositions.firstEntry().getKey();
-                    PositionImpl deletePosition = metadataPositions.firstEntry().getKey();
+                    PositionImpl firstPosition = PositionImpl.EARLIEST;
```
Comment (gaoran10) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/MLPendingAckStore.java:244]: I'm not sure why need `firstPosition`, maybe we can initialize the `deletePosition` as a null value, after checking, if the `deletePosition` is not null, then we can call the method `cursor.asyncMarkDelete`.

Comment (liangyepianzhou) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/MLPendingAckStore.java:244]: I think you are right.

Comment (liangyepianzhou) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/MLPendingAckStore.java:244]: They are equivalent

**Review by gaoran10 — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java [hunk 2]
```diff
@@ -898,4 +899,74 @@ public void testAutoCreateSchemaForTransactionSnapshot() throws Exception {
...
+                .topic(topic)
+                .sendTimeout(0, TimeUnit.SECONDS)
+                .create();
+
+        Consumer<byte[]> consumer = pulsarClient
+                .newConsumer()
+                .topic(topic)
+                .subscriptionName("sub")
+                .subscribe();
+        consumer.getSubscription();
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java:917]: The producer and consumer should close after the test.

**Pull request merged**